### PR TITLE
Add the ability to configure repository credentials for the ECS deployment

### PR DESCRIPTION
**Pull request by bruce-y**

### CODE REVIEW

File: modules/ecs/main.tf [hunk 1]
```diff
@@ -1,5 +1,31 @@
 locals {
   policies = concat([aws_iam_policy.superblocks_agent_policy.arn], var.additional_ecs_execution_task_policy_arns)
+  ecs_container_definition = {
+    name      = "superblocks-agent"
```
Comment (taha-au) [inline modules/ecs/main.tf:4]: this should probs be ${var.name_prefix}-agent`

File: modules/ecs/main.tf [hunk 2]
```diff
@@ -1,5 +1,31 @@
...
+        hostPort      = var.container_port
+      }
+    ],
+    logConfiguration = {
+      logDriver = "awslogs"
+      options = {
+        awslogs-create-group  = "true"
+        awslogs-group         = "/superblocks-agent"
+        awslogs-region        = var.region
+        awslogs-stream-prefix = "superblocks-agent"
```
Comment (taha-au) [inline modules/ecs/main.tf:24]: ditto about using the prefix

File: main.tf [hunk 3]
```diff
@@ -65,8 +65,9 @@ module "ecs" {
   additional_ecs_execution_task_policy_arns = var.additional_ecs_execution_task_policy_arns
   task_role_arn                             = var.superblocks_agent_role_arn
 
-  container_port  = local.superblocks_http_port
-  container_image = var.superblocks_agent_image
+  container_port                                = local.superblocks_http_port
+  container_image                               = var.superblocks_agent_image
+  repository_credentials_secret_arn = var.superblocks_repository_credentials_secret_arn
```
Comment (taha-au) [inline main.tf:70]: run fmt

**Review by taha-au — APPROVED** [review]

File: modules/ecs/main.tf (hunk 1 above)
Comment (bruce-y) [reply modules/ecs/main.tf:4]: I don't want to change this now since this actually would affect people using this module currently.

**Pull request merged**